src/server/replica.cc

```diff
@@ -812,7 +812,9 @@ void DflyShardReplica::StableSyncDflyReadFb(Context* cntx) {
   io::PrefixSource ps{prefix, Sock()};
 
   JournalReader reader{&ps, 0};
-  TransactionReader tx_reader{use_multi_shard_exe_sync_, journal_rec_executed_};
+  DCHECK_GE(journal_rec_executed_, 1u);
+  TransactionReader tx_reader{use_multi_shard_exe_sync_,
+                              journal_rec_executed_.load(std::memory_order_relaxed) - 1};
```

Comment:
can this be false?
```
 795   if (auto jo = loader.journal_offset(); jo.has_value()) {                                          
 796     this->journal_rec_executed_.store(*jo);                                                         
 797   } 
```

 If so, we end up decrementing `journal_rec_executed` which is `0` and will underflow taking the value`uint::max` and end up crashing at some later point

Reply:
we do always have it in replication flow. The loader is used for loading also from a file and there we do not have the journal offset, but it does always exist in replication